tests/DataObject/DataObjectDocumentTest.php

```diff
@@ -114,6 +114,17 @@ public function testShouldIndex(): void
         $this->assertFalse($docTwo->shouldIndex());
         // Document three should NOT be indexable (canView(): false)
         $this->assertFalse($docThree->shouldIndex());
+
+        // Document should be NOT indexable (as it's value is changed to ShowInSearch: 0)
+        $dataObject = $this->objFromFixture(DataObjectFakeVersioned::class, 'one');
+        $dataObject->ShowInSearch = false;
+        $dataObject->write();
+        $this->assertFalse($docOne->shouldIndex());
+
+        // Document should be indexable (as it's value is now changed to ShowInSearch: 1)
+        $dataObject->ShowInSearch = true;
+        $dataObject->write();
+        $this->assertTrue($docOne->shouldIndex());
```

Comment:
FYI there are some changes that might require you to publish the object. try a rebase to check